custom_components/ramses_cc/__init__.py

```diff
@@ -162,47 +162,25 @@ async def svc_send_packet(call: ServiceCall) -> None:
 class RamsesEntity(Entity):
     """Base for any RAMSES II-compatible entity (e.g. Climate, Sensor)."""
 
-    entity_id: str = None  # type: ignore[assignment]
-    # _attr_assumed_state: bool = False
-    # _attr_attribution: str | None = None
-    # _attr_context_recent_time: timedelta = timedelta(seconds=5)
-    # _attr_device_info: DeviceInfo | None = None
-    # _attr_entity_category: EntityCategory | None
-    # _attr_has_entity_name: bool
-    # _attr_entity_picture: str | None = None
-    # _attr_entity_registry_enabled_default: bool
-    # _attr_entity_registry_visible_default: bool
-    # _attr_extra_state_attributes: MutableMapping[str, Any]
-    # _attr_force_update: bool
-    _attr_icon: str | None
-    _attr_name: str | None
-    _attr_should_poll: bool = True
-    _attr_unique_id: str | None = None
-    # _attr_unit_of_measurement: str | None
+    _broker: RamsesBroker
+    rf_entity: RamsesRFEntity
+
+    _attr_should_poll = False
 
     def __init__(self, broker, device) -> None:
         """Initialize the entity."""
         self.hass = broker.hass
         self._broker = broker
-        self._device = device
-
-        self._attr_should_poll = False
-
-        self._entity_state_attrs = ()
+        self.rf_entity = device
 
-        # NOTE: this is bad: self.update_ha_state(delay=5)
+        self._attr_unique_id = device.id
 
     @property
     def extra_state_attributes(self) -> dict[str, Any]:
         """Return the integration-specific state attributes."""
-        attrs = {
-            a: getattr(self._device, a)
-            for a in self._entity_state_attrs
-            if hasattr(self._device, a)
+        return {
+            ATTR_ID: self.rf_entity.id,
```

Comment:
This seemed like a useful attribute for all entities to have. `_entity_state_attrs` was unused.

Reply:
Nice one.